setuptools/tests/test_build_meta.py

```diff
@@ -103,136 +102,160 @@ def run():
-
-
-def test_prepare_metadata_for_build_wheel(build_backend):
-    dist_dir = os.path.abspath('pip-dist-info')
-    os.makedirs(dist_dir)
-
-    dist_info = build_backend.prepare_metadata_for_build_wheel(dist_dir)
-
-    assert os.path.isfile(os.path.join(dist_dir, dist_info, 'METADATA'))
-
-
-@py2_only
-def test_prepare_metadata_for_build_wheel_with_str(build_backend):
-    dist_dir = os.path.abspath(str('pip-dist-info'))
-    os.makedirs(dist_dir)
-
-    dist_info = build_backend.prepare_metadata_for_build_wheel(dist_dir)
-
-    assert os.path.isfile(os.path.join(dist_dir, dist_info, 'METADATA'))
-
-
-def test_build_sdist_explicit_dist(build_backend):
-    # explicitly specifying the dist folder should work
-    # the folder sdist_directory and the ``--dist-dir`` can be the same
-    dist_dir = os.path.abspath('dist')
-    sdist_name = build_backend.build_sdist(dist_dir)
-    assert os.path.isfile(os.path.join(dist_dir, sdist_name))
-
-
-def test_build_sdist_version_change(build_backend):
-    sdist_into_directory = os.path.abspath("out_sdist")
-    os.makedirs(sdist_into_directory)
-
-    sdist_name = build_backend.build_sdist(sdist_into_directory)
-    assert os.path.isfile(os.path.join(sdist_into_directory, sdist_name))
-
-    # if the setup.py changes subsequent call of the build meta
-    # should still succeed, given the
-    # sdist_directory the frontend specifies is empty
-    with open(os.path.abspath("setup.py"), 'rt') as file_handler:
-        content = file_handler.read()
-    with open(os.path.abspath("setup.py"), 'wt') as file_handler:
-        file_handler.write(
-            content.replace("version='0.0.0'", "version='0.0.1'"))
-
-    shutil.rmtree(sdist_into_directory)
-    os.makedirs(sdist_into_directory)
-
-    sdist_name = build_backend.build_sdist("out_sdist")
-    assert os.path.isfile(
-        os.path.join(os.path.abspath("out_sdist"), sdist_name))
-
-
-def test_build_sdist_setup_py_exists(tmpdir_cwd):
-    # If build_sdist is called from a script other than setup.py,
-    # ensure setup.py is include
-    build_files(defns[0])
-    targz_path = build_sdist("temp")
-    with tarfile.open(os.path.join("temp", targz_path)) as tar:
-        assert any('setup.py' in name for name in tar.getnames())
-
-
-def test_build_sdist_setup_py_manifest_excluded(tmpdir_cwd):
-    # Ensure that MANIFEST.in can exclude setup.py
-    files = {
-        'setup.py': DALS("""
-    __import__('setuptools').setup(
-        name='foo',
-        version='0.0.0',
-        py_modules=['hello']
-    )"""),
-        'hello.py': '',
-        'MANIFEST.in': DALS("""
-    exclude setup.py
-    """)
-    }
-
-    build_files(files)
-    targz_path = build_sdist("temp")
-    with tarfile.open(os.path.join("temp", targz_path)) as tar:
-        assert not any('setup.py' in name for name in tar.getnames())
-
-
-def test_build_sdist_builds_targz_even_if_zip_indicated(tmpdir_cwd):
-    files = {
+class TestBuildMetaBackend:
```

Comment:
I would have these two in separate files to represent how their implementation is separate, but just a stylistic comment feel free to disregard.

Reply:
I did consider this, but I am re-using all the tests from `build_meta` in `build_meta_legacy` and it felt weird to import a test case class from a test file.